packages/cubejs-schema-compiler/adapter/BaseQuery.js

```diff
@@ -592,10 +592,10 @@ class BaseQuery {
             const timeSeries = d.timeSeries();
             return f(
               isFromStartToEnd ?
-                this.dateTimeCast(this.paramAllocator.allocateParam(timeSeries[0][0])) :
```

Comment:
It seems isn't equivalent to `d.dateFrom()`.

Reply:
Deepened into the code and found the reason for the differences values for date filter: 
- queries with timeSeries generated with timezone in SQL
- queries with filter by inDateRage generated date values with TZ offset 

Now I understand that the code works correctly, i revert changes and added some tests, maybe they'll be useful ))

Reply:
@RusovDmitriy Makes sense. Do you want to use this PR to fix the original issue or are you thinking about starting the new one?

Reply:
@paveltiunov In original issue was about different date format between dateRange and timeDimension, but i think is over anymore:

1. For filter with date operators, after fix https://github.com/cube-js/cube.js/pull/999 we'll get the date format like `2017-01-01T08:00:00Z`, with timezone offset

2. For timeDimension don't need any fix... yes, values have diff format like `YYYY-MM-DDT00:00:00.000`, but timeDimension build SQL with diff code and have special logic for `granularity`.

So... if you have any idea - i accept advice))

PS: I think deep down the problem is that there are no contracts in your code, DRY principles are being violated... api-gateway is a good candidate for TypeScript, it is small for fast rewrite and very important because it is public interface. Tell me if this contribution would be helpful.